### PR TITLE
Adding ftgl

### PR DESCRIPTION
**Pull request by henryiii**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Main purpose is to unbundle the dependency from ROOT. This is a fork of the original source (last released in 2008 and unmaintained since 2011). The fork is being used in Arch and Debian, and is proposed for Spack.

### CODE REVIEW

**Comment by henryiii** [comment]
@chrisburr, I'm getting `error: GL library could not be found, please specify its location with --with-gl-lib.` I assume it's at `/home/conda/feedstock_root/build_artifacts/root_1575672172253/_build_env/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib64/libGL.so`, but not sure how to get that path.

**Comment by chrisburr** [comment]
How about `$CONDA_BUILD_SYSROOT/usr/lib64/libGL.so`?

**Comment by henryiii** [comment]
Doesn't seem to be able to find it still. Is there a missing requirement, perhaps? Also, why does the line above say "Darwin specific"?

**Comment by henryiii** [comment]
Windows was provided as VS project files, the latest of which was 2008. Actually, Debian is using https://github.com/frankheckenbach/ftgl, which has a proper, updated build system. Would ROOT be able to use that if we use that one? It's on version 2.4 now.

**Comment by henryiii** [comment]
I have a version that works (except it produces a .lib instead of a .dll on windows) using the fork that Debian uses (in another branch atm). The 10 year old one still has issues finding GL and does not easily build on windows.

**Comment by chrisburr** [comment]
> I have a version that works (except it produces a .lib instead of a .dll on windows) using the fork that Debian uses (in another branch atm). The 10 year old one still has issues finding GL and does not easily build on windows.

I'm happy with that. If you push it here I'll merge and see if ROOT can handle it.

File: recipes/ftgl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    # Required for OpenGL support
+    # See https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=cdt#libgl
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
```
Comment (chrisburr) [inline recipes/ftgl/meta.yaml]: This package should provide `libGL.so`:
```bash
$ find /opt/conda/pkgs -name libGL.so
/opt/conda/pkgs/mesa-libgl-devel-cos6-x86_64-11.0.7-h5a2c0ac_0/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib64/libGL.so
```

File: recipes/ftgl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{% set name = "ftgl" %}
+{% set version = "2.4.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: "https://github.com/frankheckenbach/ftgl/archive/v2.4.0.tar.gz"
```
Comment (chrisburr) [inline recipes/ftgl/meta.yaml]:
```suggestion
  url: "https://github.com/frankheckenbach/ftgl/archive/v{{ version }}.tar.gz"
```

File: recipes/ftgl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  host:
+    - xorg-libxfixes  # [linux]
+    - libglu  # [linux]
+    - freetype
+    - libglu  # [linux]
+
+  run:
+    - libglu  # [linux]
+    - freetype
+    - libglu  # [linux]
```
Comment (chrisburr) [inline recipes/ftgl/meta.yaml]:
```suggestion
```

File: recipes/ftgl/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux]
+  host:
+    - xorg-libxfixes  # [linux]
+    - libglu  # [linux]
+    - freetype
+    - libglu  # [linux]
```
Comment (chrisburr) [inline recipes/ftgl/meta.yaml]: Duplicated
```suggestion
```

File: recipes/ftgl/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  run:
+    - libglu  # [linux]
+    - freetype
+    - libglu  # [linux]
+            
+
+test:
+   commands:
+     - test -f $PREFIX/lib/libftgl$SHLIB_EXT  # [unix]
+     - ls %LIBRARY_LIB% # [win]
```
Comment (chrisburr) [inline recipes/ftgl/meta.yaml]:
```suggestion
```

**Comment by henryiii** [comment]
Okay. Looks like Spack is working on moving to this form, and ROOT works with 2.4.0. https://github.com/spack/spack/pull/14221

Any idea why Windows is producing .lib instead of .dll, even with `BUILD_SHARED_LIBS` on?

@frankheckenbach, would you like to be listed as a maintainer?

File: recipes/ftgl/meta.yaml (hunk 3 above)
Comment (henryiii) [reply recipes/ftgl/meta.yaml]: This is needed to add `GL/glu.h`

Comment (henryiii) [reply recipes/ftgl/meta.yaml]: (I can remove it and retry to verify, but something was failing until I added this, either the CMake search or GL/glu.h)

Comment (henryiii) [reply recipes/ftgl/meta.yaml]: Oh, you mean it's (actually) duplicated

Comment (henryiii) [reply recipes/ftgl/meta.yaml]: Bad merge when rebasing.

**Comment by henryiii** [comment]
Maybe a static lib is okay on Windows, since the others are?

```
(%PREFIX%) %SRC_DIR%>ls %PREFIX%\Library\lib 
cmake
freetype.lib
ftgl.lib
libpng
libpng.lib
libpng16.lib
libpng16_static.lib
libpng_static.lib
pkgconfig
z.lib
zdll.lib
zlib.lib
zlibstatic.lib
```

?

**Comment by henryiii** [comment]
Arch also uses this fork, too. https://pkgs.org/download/ftgl

File: recipes/ftgl/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+test:
+   commands:
+     - test -f $PREFIX/lib/libftgl$SHLIB_EXT  # [unix]
+     - if not exist %LIBRARY_LIB%\ftgl.lib exit 1  # [win]
+
+about:
+  home: https://github.com/frankheckenbach/ftgl
+  license: LGPL-2.0-or-later
+  license_family: MIT
```
Comment (chrisburr) [inline recipes/ftgl/meta.yaml]:
```suggestion
  license_family: LGPL
```

**Comment by chrisburr** [comment]
> Maybe a static lib is okay on Windows, since the others are?

That's enough to persuade me that it's okay 👍

Excluding the minor metadata correction, I think this is ready to merge?

File: recipes/ftgl/meta.yaml (hunk 6 above)
Comment (henryiii) [reply recipes/ftgl/meta.yaml]: Actually, this is MIT, I just need to fix the one above instead.

**Comment by chrisburr** [comment]
LGTM 👍

Thanks (and happy new year) @henryiii!!! 🍰

**Pull request merged**